src/seedu/addressbook/data/person/Address.java

```diff
@@ -25,10 +30,21 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
         if (!isValidAddress(address)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
-        this.value = address;
+        setAddressFields(address);
     }
 
     /**
+     * Split the address into various fields
+     */
+	private void setAddressFields(String address) {
+		addressComponents = address.split(",",-1);
```

Comment:
Indentation problem here?. Our coding standard requires you to use 4 spaces instead of tabs. You can configure Eclipse to convert tabs to spaces automatically.